packages/sdks/src/functions/extract-localized-values.ts

```diff
@@ -0,0 +1,58 @@
+import type { BuilderBlock } from '../types/builder-block.js';
+
+function isLocalizedField(value: any) {
+  return (
+    value &&
+    typeof value === 'object' &&
+    value['@type'] === '@builder.io/core:LocalizedValue'
+  );
+}
+
+function containsLocalizedValues(data: Record<string, any>) {
+  if (!data || !Object.getOwnPropertyNames(data).length) {
+    return false;
+  }
+  return Object.values(data).some(isLocalizedField);
+}
+
+function extractLocalizedValues(data: Record<string, any>, locale: string) {
+  if (!data || !Object.getOwnPropertyNames(data).length) {
+    return {};
+  }
+  const extractedResult: Record<string, any> = {};
+
+  const ownKeys = Object.getOwnPropertyNames(data);
+  for (const key of ownKeys) {
+    const value = data[key];
+    extractedResult[key] = isLocalizedField(value) ? value[locale] : value;
+  }
+  return extractedResult;
+}
+
+export function resolveLocalizedValues(
+  block: BuilderBlock,
+  locale: string | undefined
+) {
+  if (
+    block.component?.options &&
+    containsLocalizedValues(block.component?.options) &&
+    !locale
+  ) {
+    console.warn(
+      '[Builder.io] In order to use localized fields in Builder, you must pass a locale prop to the BuilderComponent or to options object while fetching the content to resolve localized fields. Learn more: https://www.builder.io/c/docs/localization-inline#targeting-and-inline-localization'
+    );
+  }
+
+  if (
+    block.component?.options &&
+    containsLocalizedValues(block.component?.options) &&
+    locale
+  ) {
+    block.component.options = extractLocalizedValues(
```

Comment:
I think you'd want to use 'traverse' here, to account for sub fields

Reply:
thanks for the catch, will update the logic